### PR TITLE
Grad School Statement of Purpose

### PR DESCRIPTION
**Pull request by theNerd247**
I've written this using Conix - with that said the `pdf "sop" [ ... ]` is the actual content of the statement (lines 24 - 79). So any lines that are commented out should be ignored - I've kept them around to help act as a drafting space (maybe they'll give some hint as to what's going on in my mind).

### CODE REVIEW

File: notes/gradSchoolPurposeStatement.nix [hunk 1]
```diff
@@ -1,146 +1,257 @@
...
+# If you are applying to more than one program, you may (but are not required
+# to) submit a separate Statement of Purpose for each program. If you are
+# submitting different statements, please upload as one file and include a
+# table of contents page. Include your name and User ID on the essay
+
+conix: with conix; [ (pdf "sop" [''
+
+As a PhD student at Carnegie Mellon I intend to study how to use mathematical
+models to design and implement domain specific languages.  Particularly I am
+interested in using category theory, Type Theory, and the lambda calculi to
```
Comment (Gabriella439) [inline notes/gradSchoolPurposeStatement.nix]:
```suggestion
interested in using category theory, type theory, and the lambda calculi to
```

File: notes/gradSchoolPurposeStatement.nix [hunk 2]
```diff
@@ -1,146 +1,257 @@
...
+directly tied to the existence of non-idiomatic libraries that did not have
+well defined behavior. Since then I've learned the enormous benefit of model
+driven development. I began to study category theory and lambda calculus in
+order to better understand how programming languages were constructed in hopes
+of discovering better techniques for designing program APIs and applications.
+I have come to realize that most APIs are small languages and
+applications are compositions of statements written in those languages. Since
+APIs are languages and good languages have some underlying mathematical model
+it only makes sense to study languages and their modelling to then develop
+industrial strength APIs.
```
Comment (Gabriella439) [inline notes/gradSchoolPurposeStatement.nix]:
```suggestion
it only makes sense to study languages and their modeling to then develop
industrial strength APIs. 
```

File: notes/gradSchoolPurposeStatement.nix [hunk 3]
```diff
@@ -1,146 +1,257 @@
...
+# opportunity to work on a project involving a robotic arm weilding a knife
+# blade to debone chicken carcasses. The program was written largely in C++ as
+# well as in a propriatary language akin to a crude FORTRAN. The program
+# involved everything from 3-d image processing, multi-threading, memory
+# management and high-level process logic. Testing was a nightmare and due to a
+# tight budget time couldn't be spent on properly following programming best
+# practices.  However, safety was a large factor in how the code was designed
+# as well as in testing. While we didn't have one then a proper mathematical
+# models and an eDSL would have made the project cost less and involve fewer
+# broken parts.  Thankfully no humans were physically injured - just a lot of
```
Comment (Gabriella439) [inline notes/gradSchoolPurposeStatement.nix:92]: You should probably be more specific here about how an eDSL would have helped

**Review by Gabriella439 — COMMENTED** [review]
This looks good so far.  I think you should spend more time being more specific on how programming languages / DSLs / eDSLs would help the domains that you are talking about.  You're doing fine with respect to covering how you are well-suited to pursue these topics.

**Comment by theNerd247** [comment]
Thanks Gabe :) I'll work on those next. So should this mostly be a bit of my arguments / an advocation for the areas I wish to study?

**Pull request closed without merging**